spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialWithinChannelLRN.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.nn.abstractnn.TensorModule
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+/**
+ * The local response normalization layer performs a kind of “lateral inhibition”
+ * by normalizing over local input regions. the local regions extend spatially,
+ * in separate channels (i.e., they have shape 1 x local_size x local_size).
+ * @param size  the side length of the square region to sum over
+ * @param alpha the scaling parameter
+ * @param beta the exponent
+ * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
+ */
+class SpatialWithinChannelLRN[@specialized(Float, Double) T: ClassTag]
```

Comment:
Remove specialized to avoid creating multiple class objects, since we use the factory method (where we have 'specialzied') to create instance, there is no type inference cost. Also this is required in module serialization :)